### PR TITLE
Simplify some naming

### PR DESCRIPTION
**Pull request by mdiep**
I really dislike some of the current naming in ReactiveTask. This PR cleans that up and makes a few more breaking changes while I'm at it:

- Rename `TaskDescription` to `Task`
- Remove the `launchPath:` parameter name from `Task.init`
- Rename `ReactiveTaskError` to `TaskError`
- Move `standardInput` from `Task` to `launchTask`
- Make `Task` conform to `Hashable`
- Add a `Launch` case to `TaskEvent` that sends the `Task`

The latter 3 changes were motivated by a desire to print executed commands in Carthage. Right now, we only print the output of commands. It'd be nice to print the commands themselves.

### CODE REVIEW

File: ReactiveTask/Task.swift [hunk 1]
```diff
@@ -48,18 +42,46 @@ public struct TaskDescription {
...
+			result ^= argument.hashValue
 		}
+		for (key, value) in environment ?? [:] {
+			result ^= key.hashValue ^ value.hashValue
+		}
+		return result
 	}
 }
 
+private func ==<Key : Equatable, Value : Equatable>(lhs: [Key : Value]?, rhs: [Key : Value]?) -> Bool {
```
Comment (ikesyo) [inline ReactiveTask/Task.swift]: This should be styled as follows:

```swift
<Key: Equatable, Value: Equatable>(lhs: [Key: Value]?, rhs: [Key: Value]?)
```

File: ReactiveTask/Task.swift [hunk 2]
```diff
@@ -48,18 +42,46 @@ public struct TaskDescription {
...
+	switch (lhs, rhs) {
+	case let (.Some(lhs), .Some(rhs)):
```
Comment (ikesyo) [inline ReactiveTask/Task.swift]: This could be written as `case let (lhs?, rhs?):`.

File: ReactiveTask/Task.swift [hunk 3]
```diff
@@ -391,14 +428,18 @@ extension Signal where Value: TaskEventType {
 	}
 }
 
-/// Launches a new shell task, using the parameters from `taskDescription`.
+/// Launches a new shell task.
+///
+/// taskDescription - The task to launch.
```
Comment (ikesyo) [inline ReactiveTask/Task.swift]: How about using the Xcode 7's markdown syntax:

```swift
/// Launches a new shell task.
///
/// - parameter taskDescription: The task to launch.
/// - parameter standardInput:   Data to stream to standard input of the launched process. If nil, stdin will
///                              be inherited from the parent process.
///
/// - returns: A producer that will launch the task when started, then send
///            `TaskEvent`s as execution proceeds.
```

**Comment by ikesyo** [comment]
I've experimented with the changes in Carthage, and I had no problem to put them in! Looks good to me and just a few stylistic notes. :+1:

File: ReactiveTask/Task.swift (hunk 2 above)
Comment (mdiep) [inline ReactiveTask/Task.swift]: TIL!

**Comment by mdiep** [comment]
:space_invader:

File: ReactiveTask/Task.swift [hunk 4]
```diff
@@ -391,14 +428,19 @@ extension Signal where Value: TaskEventType {
 	}
 }
 
-/// Launches a new shell task, using the parameters from `taskDescription`.
+/// Launches a new shell task.
+///
+/// - Parameters:
```
Comment (ikesyo) [inline ReactiveTask/Task.swift]: It's nice! :100:

Comment (mdiep) [inline ReactiveTask/Task.swift]: I just discovered this yesterday [in the apple/swift documentation](https://github.com/apple/swift/blob/master/docs/DocumentationComments.md). :smile:

**Pull request merged**